common/unit-test/CMakeLists.txt

```diff
@@ -0,0 +1,101 @@
+
+# Copyright (C) 2016 Pelagicore AB
+#
+# Permission to use, copy, modify, and/or distribute this software for
+# any purpose with or without fee is hereby granted, provided that the
+# above copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+# WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+# BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+# OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+# ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+# SOFTWARE.
+#
+# For further information see LICENSE
+
+
+
+set(TEST_LIBRARY_DEPENDENCIES 
+    ${DBusCpp_LIBRARIES}
+    ${Glibmm_LIBRARIES}
+    ${LXC_LIBRARIES}
+    ${Jansson_LIBRARIES}
+    ${IVILogging_LIBRARIES}
+    softwarecontainerlib
+)
+include_directories(${SOFTWARECONTAINER_COMMON_DIR})
+
+set(TEST_FILES
+    softwarecontainer-common_unittest.cpp
+    ../softwarecontainer-common.cpp
+    main.cpp
+)
+
+macro(add_gtest_test testName files libraries)
+
+	if(NOT DEFINED TESTS_INSTALLATION_PATH)
+		set(TESTS_INSTALLATION_PATH "lib/${PROJECT_NAME}/ptest")
+	endif()
+
+	message("Tests will be installed in : ${TESTS_INSTALLATION_PATH}")
+
+	if (DEFINED $ENV{GTEST_DIR})
+		set(GTEST_DIR $ENV{GTEST_DIR})
+	endif()
+
+	message("Google test found in ${GTEST_DIR}")
+
+	include_directories(${GTEST_DIR}/include)
+	link_directories(${GTEST_DIR}/lib/.libs)
+
+	add_executable(${testName}
+		${files}
+	)
+
+	target_link_libraries(${testName}
+		${libraries}
+		gtest
+		gtest_main
+		gmock
+		gmock_main
+	)
+
+	INSTALL(TARGETS ${testName}
+	  DESTINATION ${TESTS_INSTALLATION_PATH}
+	)
+
+	add_test(
+	    NAME ${testName}
+	    COMMAND ${testName}
+	)
+
+	file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/run-ptest "#!/bin/bash
+FOLDER=`dirname $0`
+for filename in $FOLDER/*; do
+	TEST_FILE=$filename
+	if [ \"`basename $filename`\" != \"run-ptest\" ]; then
+		if [[ -x $TEST_FILE ]]; then
+#			echo Starting $TEST_FILE
+			TMP_FILE=`mktemp`
+			# Run the test and convert its output to the ptest format
+			$TEST_FILE >> $TMP_FILE
+			cat $TMP_FILE | sed -r 's/^\\[\\s+OK\\s+\\] (.*) \\([0-9]+\\sms\\)$/PASS: \\1 /' | sed -r 's/^\\[\\s+FAILED\\s+\\] (.*) \\([0-9]+\\sms\\)$/FAIL: \\1 /' | awk '{if ($1 == \"PASS:\" || $1 == \"FAIL:\") {print $0}}'
```

Comment:
This is a copy paste. we should probably centralize the entire macro also. It's probably the third time i did the copy...